src/python/pants/rules/core/test_test.py

```diff
@@ -4,34 +4,60 @@
 import logging
 from textwrap import dedent
 from typing import Dict, Optional
+from unittest.mock import Mock
 
-from pants.base.specs import DescendantAddresses, SingleAddress, Spec
+from pants.base.specs import DescendantAddresses, SingleAddress, Spec, Specs
 from pants.build_graph.address import Address, BuildFileAddress
 from pants.engine.build_files import AddressProvenanceMap
 from pants.engine.fs import EMPTY_DIRECTORY_DIGEST, Snapshot
 from pants.engine.legacy.graph import HydratedTarget
 from pants.engine.legacy.structs import PythonBinaryAdaptor, PythonTestsAdaptor
 from pants.engine.rules import UnionMembership
-from pants.rules.core.core_test_model import Status, TestResult, TestTarget
-from pants.rules.core.test import AddressAndTestResult, coordinator_of_tests, fast_test
+from pants.rules.core.core_test_model import Status, TestDebugResult, TestResult, TestTarget
+from pants.rules.core.test import (
+  AddressAndDebugResult,
+  AddressAndTestResult,
+  coordinator_of_tests,
+  run_tests,
+)
 from pants.source.wrapped_globs import EagerFilesetWithSpec
 from pants.testutil.engine.util import MockConsole, MockGet, run_rule
 from pants.testutil.test_base import TestBase
 
 
+class MockOptions:
+  def __init__(self, **values):
+    self.values = Mock(**values)
+
+
 class TestTest(TestBase):
-  def single_target_test(self, result, expected_console_output, success=True):
+  def single_target_test(self, result, expected_console_output, success=True, debug=False):
     console = MockConsole(use_colors=False)
+    options = MockOptions(debug=debug)
     addr = self.make_build_target_address("some/target")
     res = run_rule(
-      fast_test,
-      rule_args=[console, (addr,)],
+      run_tests,
+      rule_args=[console, options, (addr,)],
       mock_gets=[
         MockGet(
```

Comment:
Can you conditionally set `mock_gets` to either the debug or non-debug `Get`s based on the value of `debug` so that we validate in the tests that we don't call into the wrong ones?

Reply:
This definitely feels like it'd be helpful here, though AFAICT we can't (currently) condition `mock_gets` in tests as `run_rule` uses `task_rule.input_gets`, which itself is a flat list put together in the `_RuleVisitor` of all `Get`s in a rule body (and doesn't account for conditionals).